src/robot/parsing/model/statements.py

```diff
@@ -986,17 +991,30 @@ def condition(self):
 
     @property
     def limit(self):
-        value = self.get_value(Token.OPTION)
-        return value[len('limit='):] if value else None
+        values = self.get_values(Token.OPTION)
+        for value in values:
+            if value.startswith('limit='):
+                return value[len('limit='):]
+        return None
```

Comment:
I added `Statement.get_option()` to ease getting options with `FOR IN ZIP` loop and updated this code as well so that it is nowadays just:
```python
@property
def limit(self):
    return self.get_option('limit')
```
The change has obviously caused a conflict, sorry for that, but the good news is that the `on_limit_message` property can be implemented using the same approach.